FauxPose/plugin.cpp

```diff
@@ -0,0 +1,185 @@
+		return false;
+	}
+
+	virtual pose_type correct_pose([[maybe_unused]] const pose_type pose) const override {
+		pose_type simulated_pose;
+		printf("[fauxpose] Returning (zero) pose\n");
+		return simulated_pose;
+	}
+
+	// ********************************************************************
+	virtual Eigen::Quaternionf get_offset() override {
+		return offset;
+	}
+
+	virtual void set_offset(const Eigen::Quaternionf& raw_o_times_offset) override{
+		std::unique_lock lock {offset_mutex};
+		Eigen::Quaternionf raw_o = raw_o_times_offset * offset.inverse();
+		//std::cout << "pose_prediction: set_offset" << std::endl;
+		offset = raw_o.inverse();
+	}
+
+	// ********************************************************************
+	virtual fast_pose_type get_fast_pose() const override  {                                          
+	//	const switchboard::ptr<const switchboard::event_wrapper<time_type>> estimated_vsync = _m_vsync_estimate.get_ro_nullable();
+	//	if (estimated_vsync == nullptr) {                                                            
+	//		std::cerr << "Vsync estimation not valid yet, returning fast_pose for now()" << std::endl;
+			return get_fast_pose(std::chrono::system_clock::now());                                  
+	//	} else {                                                                                     
+	//		return get_fast_pose(**estimated_vsync);                                                 
+	//	}                                                                                            
+	}
+
+	// ********************************************************************
+	// get_fast_pose(): returns a "fast_pose_type" with the algorithmically
+	//   determined location values.  (Presently moving in a circle, but
+	//   always facing "front".)
+	//
+	// NOTE: time_type == std::chrono::system_clock::time_point
+	virtual fast_pose_type get_fast_pose(time_type time) const override {
+		pose_type simulated_pose;
+		double	sim_time;
+
+		RAC_ERRNO_MSG("[fauxpose] at start of _p_one_iteration");
+
+		// Calculate simulation time from start of execution
+		std::chrono::nanoseconds elapsed_time;
+		//auto now = std::chrono::system_clock::now();
+		//time_type current_time = std::chrono::time_point_cast<std::chrono::nanoseconds>(now);
+		//elapsed_time = current_time - sim_start_time;
+		elapsed_time = time - sim_start_time;
+		sim_time = elapsed_time.count() * 0.000000001;
+
+		//printf("[fauxpose] time = %ld\n", std::chrono::high_resolution_clock::now().time_since_epoch().count());
+		printf("[fauxpose] elapsed time = %ld (%lf)\n", elapsed_time.count(), sim_time);
+		//fprintf(stderr, "[fauxpose] ********* elapsed time = %ld (%lf)\n", elapsed_time.count(), sim_time);
+
+		// Calculate new pose values
+		//simulated_location = Eigen::Vector3f{0.1f, 0.0f, 0.0f};
+
+		//fprintf(stderr, "FP: time.count is %ld\n", time.time_since_epoch());
+		simulated_pose.position[0] = amplitude * sin(sim_time * period);	// X
```

Comment:
Note in a comment that `sim_start_time` is just a phase offset in the sine wave.

Reply:
Added variable descriptions here and elsewhere.